frontend/src/app/util/dom.cljs

```diff
@@ -464,11 +464,11 @@
 
 (defn trigger-download-uri
   [filename mtype uri]
-  (let [link (create-element "a")
+  (let [link      (create-element "a")
         extension (mtype->extension mtype)
-        filename (if extension
-                   (str filename "." extension)
-                   filename)]
+        filename  (if (and extension (not (str/ends-with? filename extension)))
+                    (str/concat filename "." extension)
```

Comment:
If the extension already has a `.`, no need to add it here again.

Reply:
fixed